ApplicationLibCode/ProjectDataModel/WellPath/RimFixedTrajectoryWellPath.cpp

```diff
@@ -0,0 +1,113 @@
+///
+//--------------------------------------------------------------------------------------------------
+RimFixedTrajectoryWellPath::RimFixedTrajectoryWellPath()
+{
+    CAF_PDM_InitScriptableObject( "Fixed Trajectory Well Path", ":/Well.svg", "", "FixedTrajectoryWellPath" );
+
+    CAF_PDM_InitScriptableField( &m_wellTargets, "WellTargets", std::vector<cvf::Vec3d>(), "Well Targets" );
+    m_wellTargets.uiCapability()->setUiHidden( true );
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+RimFixedTrajectoryWellPath::~RimFixedTrajectoryWellPath()
+{
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+void RimFixedTrajectoryWellPath::setWellTargets( const std::vector<cvf::Vec3d>& wellTargets )
+{
+    m_wellTargets = wellTargets;
+    createWellPathGeometry();
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+std::vector<cvf::Vec3d> RimFixedTrajectoryWellPath::wellTargets() const
+{
+    return m_wellTargets();
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+void RimFixedTrajectoryWellPath::createWellPathGeometry()
+{
+    if ( m_wellTargets().empty() )
+    {
+        return;
+    }
+
+    // Create well path geometry from the target points
+    std::vector<cvf::Vec3d> wellPathPoints = m_wellTargets();
+    std::vector<double>     measuredDepths;
+
+    // Calculate measured depths along the path
+    double totalMD = 0.0;
+    measuredDepths.push_back( totalMD );
+
+    for ( size_t i = 1; i < wellPathPoints.size(); ++i )
+    {
+        cvf::Vec3d segmentVector = wellPathPoints[i] - wellPathPoints[i - 1];
+        double     segmentLength = segmentVector.length();
+        totalMD += segmentLength;
+        measuredDepths.push_back( totalMD );
+    }
+
+    auto wellPathGeometry = new RigWellPath( wellPathPoints, measuredDepths );
+
+    setWellPathGeometry( wellPathGeometry );
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+void RimFixedTrajectoryWellPath::defineUiOrdering( QString uiConfigName, caf::PdmUiOrdering& uiOrdering )
+{
+    RimWellPath::defineUiOrdering( uiConfigName, uiOrdering );
+
+    // Hide the well targets field from UI since this is Python-only
```

Comment:
I think this comment is wrong, RimFixedTrajectoryWellPath derives from RimWellPath

Reply:
The points are called m_wellTargets in this class, too. 

Not to be confused with the RimModeledWellPath::m_wellTargets.